Cython/Compiler/ModuleNode.py

```diff
@@ -233,8 +233,32 @@ def h_entries(entries, api=0, pxd=0):
                 h_code.putln('#error "Unicode module names are not supported in Python 2";')
             h_code.putln("PyMODINIT_FUNC init%s(void);" % py2_mod_name)
             h_code.putln("#else")
-            h_code.putln("PyMODINIT_FUNC %s(void);" % self.mod_init_func_cname('PyInit', env))
-            h_code.putln("#endif")
+            py3_mod_func_name = self.mod_init_func_cname('PyInit', env)
+            warning_string = EncodedString('Use PyImport_AppendInittab("%s", %s) instead of calling %s directly.' % (
+                py2_mod_name, py3_mod_func_name, py3_mod_func_name))
+            h_code.putln('/* WARNING: %s from Python 3.5 */' % warning_string.rstrip('.'))
+            h_code.putln("PyMODINIT_FUNC %s(void);" % py3_mod_func_name)
+            h_code.putln("")
+            h_code.putln("#if PY_MAJOR_VERSION == 3 && PY_MINOR_VERSION >= 5 "
```

Comment:
Let's tie this to the actual feature flag that makes the call invalid.
```suggestion
            h_code.putln("#if CYTHON_PEP489_MULTI_PHASE_INIT "
```

Reply:
I don't think that makes it into the header file though?

Reply:
Argh – right. Thanks for catching that. :) I'll change it back.